operator/internal/custom/istio_client.go

```diff
@@ -15,46 +17,84 @@ import (
 )
 
 const (
-	firstElementIdx = 0
-	contractVersion = "1"
+	firstElementIdx    = 0
+	contractVersion    = "1"
+	virtualServiceName = "kcp-events"
 )
 
 type IstioClient struct {
 	istioclient.Interface
+	// IstioGateway string
+	IstioGateway client.ObjectKey
 }
 
-func NewVersionedIstioClient(cfg *rest.Config) (*IstioClient, error) {
+func NewVersionedIstioClient(cfg *rest.Config, istioGw string) (*IstioClient, error) {
+	gwStrings := strings.Split(istioGw, "/")
+	if len(gwStrings) != 2 {
+		return nil, fmt.Errorf("error validating gateway config: expected (namespace/name), got: (%s)", istioGw)
```

Comment:
Can we assume this Gateway is always deployed in `default` namespace with name `kyma-gateway` (or consider other name to avoid potential naming conflict)? How about fix the name and namespace, what will be the side effect? Then we can get rid of this parsing logical.

Reply:
I think the gateway name should in general be discovered and if there are more than one, we should use a label selector for getting the correct gateway. Reason for this is that I think it can vary in name between the different KCP environments.

Reply:
Hi @jakobmoellersap, I think you're right. I have created a ticket for your suggestion:
#234 
Please have a look and check if the ACs are to the point 👍

Reply:
Added a suggestion to the AC